### ISSUE TITLE
Fix infinite type check to not generate false positives

### ISSUE THREAD

**Issue by andrew4328** [Lambda-Mountain-Compiler-Backend/lambda-mountain#1419]
```
type A = A { v:Vector<A> };
```

would brick.

**Comment by andrew4328** [comment]
I gave this a hard-coded workaround for now, but this needs to be fixed.
